pkg/localstore/mode_put_test.go

```diff
@@ -484,79 +485,56 @@ func TestModePut_sameChunk(t *testing.T) {
 // TestModePut_sameChunk puts the same chunk multiple times
 // and validates that all relevant indexes have only one item
 // in them.
-func TestModePut_sameStamp(t *testing.T) {
+func TestModePut_SameStamp(t *testing.T) {
+
 	ctx := context.Background()
-	modes := []storage.ModePut{storage.ModePutRequest, storage.ModePutRequestPin}
-	for _, mode1 := range modes {
-		for _, mode2 := range modes {
-			t.Run("chunk on same index - timestamps in order", func(t *testing.T) {
-				db := newTestDB(t, nil)
-				// call unreserve on the batch with radius 0 so that
-				// localstore is aware of the batch and the chunk can
-				// be inserted into the database
-				first := generateTestRandomChunk()
-				second := generateTestRandomChunk()
-				stamp := first.Stamp()
-				ts := binary.BigEndian.Uint64(stamp.Timestamp())
-				buf := make([]byte, 8)
-				binary.BigEndian.PutUint64(buf, ts+1)
-				second = second.WithStamp(postage.NewStamp(stamp.BatchID(), stamp.Index(), buf, stamp.Sig()))
-				unreserveChunkBatch(t, db, 0, first, second)
-
-				_, err := db.Put(ctx, mode1, first)
-				if err != nil {
-					t.Fatal(err)
-				}
-				_, err = db.Put(ctx, mode2, second)
-				if err != nil {
-					t.Fatal(err)
-				}
-				newItemsCountTest(db.retrievalDataIndex, 1)(t)
-				newItemsCountTest(db.postageChunksIndex, 1)(t)
-				newItemsCountTest(db.postageRadiusIndex, 1)(t)
-				newItemsCountTest(db.postageIndexIndex, 1)(t)
-				newItemsCountTest(db.pullIndex, 1)(t)
-				_, err = db.Get(ctx, storage.ModeGetLookup, second.Address())
-				if err != nil {
-					t.Fatalf("expected no error, got %v", err)
-				}
-				_, err = db.Get(ctx, storage.ModeGetLookup, first.Address())
-				if !errors.Is(storage.ErrNotFound, err) {
-					t.Fatalf("expected %v, got %v", storage.ErrNotFound, err)
-				}
-			})
-			t.Run("chunk on same index - timestamps in reverse order", func(t *testing.T) {
+	stamp := postagetesting.MustNewStamp()
+	ts := time.Now().Unix()
+
+	for _, modeTc := range []storage.ModePut{storage.ModePutRequest, storage.ModePutRequestPin, storage.ModePutSync} {
```

Comment:
In the previous tests we were using mixed modes in the tests as well. (mode1 and mode2 could be same or different)

Are these adding any value? I guess if the handling for different modes changes in the future this test might be required?